### PR TITLE
Move RunRC-like functions to test/utils

### PR DESCRIPTION
**Pull request by gmarek**
Ref. #34336

cc @timothysc - the "move" part of the small refactoring. @jayunit100

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34544)

### CODE REVIEW

File: test/utils/runners.go [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+type containerFailures struct {
+	status   *api.ContainerStateTerminated
+	Restarts int
+}
+
+// failedContainers inspects all containers in a pod and returns failure
+// information for containers that have failed or been restarted.
+// A map is returned where the key is the containerID and the value is a
+// struct containing the restart and failure information
+func FailedContainers(pod *api.Pod) map[string]containerFailures {
```
Comment (wojtek-t) [inline test/utils/runners.go]: This function now exists in two places. This is wrong.

**Comment by wojtek-t** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by wojtek-t** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by gmarek** [comment]
@wojtek-t PTAL

File: test/utils/runners.go (hunk 1 above)
Comment (gmarek) [inline test/utils/runners.go]: Outdated

File: test/utils/getters.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+/*
```
Comment (wojtek-t) [inline test/utils/getters.go]: getters are probably not the best name for this file.
maybe "conditions.go" or sth like that?

**Comment by gmarek** [comment]
@wojtek-t PTAL

**Comment by wojtek-t** [comment]
lgtm

**Pull request merged**